tests-js/protocol/ChangeLoanDurationTestToken.test.js

```diff
@@ -0,0 +1,461 @@
+			const deposit_amount_in_collateral = deposit_amount.mul(precision).div(rate);
+
+			const loanMaintenance = await LoanMaintenance.at(sovryn.address);
+			await loanMaintenance.extendLoanDuration(loan_id, deposit_amount, true, "0x", { from: borrower });
+
+			const end_loan_interest_data = await sovryn.getLoanInterestData(loan_id);
+			const end_loan = await sovryn.getLoan(loan_id);
+
+			expect(end_loan["endTimestamp"] == parseInt(initial_loan["endTimestamp"]) + days_to_extend.toNumber() * 24 * 60 * 60).to.be
+				.true;
+			expect(
+				end_loan_interest_data["interestDepositTotal"].eq(initial_loan_interest_data["interestDepositTotal"].add(deposit_amount))
+			).to.be.true;
+			// expect(new BN(end_loan["collateral"]).eq(new BN(initial_loan["collateral"]).sub(deposit_amount_in_collateral))).to.be.true;
+			// expect((await RBTC.balanceOf(sovryn.address)).eq(initial_collateral_token_lender_balance.sub(deposit_amount_in_collateral))).to
+			// 	.be.true;
+			expect((await SUSD.balanceOf(sovryn.address)).lte(initial_loan_token_lender_balance.add(deposit_amount))).to.be.true;
+		});
+
+		it("Test extend loan_duration with collateral sov token reward payment", async () => {
+			// prepare the test
+			await set_demand_curve(loanToken);
+			await lend_to_pool(loanToken, SUSD, owner);
+			const [loan_id, borrower] = await borrow_indefinite_loan(loanToken, sovryn, SUSD, RBTC, accounts);
+
+			const initial_loan_interest_data = await sovryn.getLoanInterestData(loan_id);
+
+			const days_to_extend = new BN(10);
+			const owed_per_day = initial_loan_interest_data["interestOwedPerDay"];
+			const deposit_amount = owed_per_day.mul(days_to_extend);
+
+			await increaseTime(10 * 24 * 60 * 60);
+			const borrower_initial_balance = await SOV.balanceOf(borrower);
+
+			const loanMaintenance = await LoanMaintenance.at(sovryn.address);
+			const { receipt } = await loanMaintenance.extendLoanDuration(loan_id, deposit_amount, true, "0x", { from: borrower });
+
+			const decode = decodeLogs(receipt.rawLogs, FeesEvents, "EarnReward");
+			const args = decode[0].args;
+			expect(args["receiver"] == borrower).to.be.true;
+			expect(args["token"] == SOV.address).to.be.true;
+			expect(args["loanId"] == loan_id).to.be.true;
+			expect(args["amount"]).to.eq((await SOV.balanceOf(borrower)).sub(borrower_initial_balance).toString());
+		});
+
+		it("Test extend loan_duration with collateral and eth should fail", async () => {
+			// prepare the test
+			await set_demand_curve(loanToken);
+			await lend_to_pool(loanToken, SUSD, owner);
+			const [loan_id, borrower] = await borrow_indefinite_loan(loanToken, sovryn, SUSD, RBTC, accounts);
+
+			const initial_loan_interest_data = await sovryn.getLoanInterestData(loan_id);
+
+			const days_to_extend = new BN(10);
+			const owed_per_day = initial_loan_interest_data["interestOwedPerDay"];
+			const deposit_amount = owed_per_day.mul(days_to_extend);
+
+			await increaseTime(10 * 24 * 60 * 60);
```

Comment:
there id no time travel in the original test
```suggestion
//			await increaseTime(10 * 24 * 60 * 60);
```

Reply:
fixed